assets/src/stories-editor/components/media-inserter/index.js

```diff
@@ -118,6 +132,7 @@ const applyWithSelect = withSelect( ( select ) => {
 		},
 		// As used in <HeaderToolbar> component
 		showInserter: select( 'core/edit-post' ).getEditorMode() === 'visual' && select( 'core/editor' ).getEditorSettings().richEditingEnabled,
+		mediaType: getCurrentPageBackgroundType(),
```

Comment:
Just realized that the `Shortcuts` component is using almost identical `withSelect` logic except for the added `mediaType`.

Is there a way how to combine those?
For example:
These are both functional components so we could do the following instead:
- Create a helper hook for the `withSelect` logic that matches and reuse it in both components directly, remove the `withSelect`.
- Use an additional `useSelect` for the `mediaType` within `MediaInserter` directly.

Not sure how much that makes sense here. Thoughts?

Reply:
I know that the logic is repeated here. But I am not sure of the benefit of a helper. At some point the code might change the logic might need to differ. Shared code might not be useful here. With mediaType, there is enough of a different to overlook a very small amount of repeated code here.

Reply:
Generalising this can be tricky, as it can be hard to find the right separation, but I think the common redux way would be to move the logic to selectors, and then reuse those. So this function would be:

```
import {
	selectIsReordering,
	selectCanInsertBlockType,
	selectShowInserter,
	selectMediaType,
} from './some/selector';

...

const applyWithSelect = withSelect( ( select ) => {
	const isReordering = selectIsReordering( select );
	const canInsertBlockType = selectCanInsertBlockType( select );
	const showInserter = selectShowInserter( select );
	const mediaType = selectMediaType( select );

	return { isReordering, canInsertBlockType, showInserter, mediaType };
} );
```

This does help with unit testing (as the applySelect invocation is completely trivial now and can be ignored for testing purposes, wheres the selectors are testable units).

However if it's not common practice to move the selector logic to external helpers, it makes the coding practice inconsistent.

Reply:
Thanks for chiming in, @barklund, makes sense!